numpy/ma/core.py

```diff
@@ -1815,7 +1815,7 @@ def masked_where(condition, a, copy=True):
     else:
         cls = MaskedArray
     result = a.view(cls)
-    result._mask = cond
+    result.mask = cond
```

Comment:
Could you add a comment here that we need to use the property `.mask` rather than `._mask` to assure that the mask is set correctly for structured arrays?